### PR TITLE
Add basic sharness/TAP tests for python binding

### PR DESCRIPTION
**Pull request by dongahn**
This PR adds basic functionality tests for PerfflowAspect's
python binding.

The tests are functionally identical with c-binding tests

It also includes one misc. changes to python binding
code: changing MD5 to SHA1  when a digest is needed
for an instance ID.
Because C binding uses SHA1 digest, this makes comparison
and grouping of outputs between C and python users
a bit better. 

It also adjusted python test programs with some
of the latest changes.

### CODE REVIEW

**Comment by dongahn** [comment]
@SteVwonder: can you take a quick look at it to land?

File: src/python/test/t0001-pybinding-basic.t [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+test_expect_success 'PERFFLOW_OPTIONS: SLURM_JOB_ID + STEP_ID must be given' '
+    SLURM_JOB_ID=123456 \
+PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    sha1=$(echo -n "1" | sha1sum | awk "{print \$1}" | cut -c1-8) &&
+    test -f perfflow.{${sha1}}.pfw &&
+    sanity_check perfflow.{${sha1}}.pfw &&
+    rm perfflow.{${sha1}}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: LSB_JOBID along will not work' '
```
Comment (SteVwonder) [inline src/python/test/t0001-pybinding-basic.t:107]: `along` -> `alone`

File: src/python/test/t0001-pybinding-basic.t [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    LSB_JOB_ID=123456 \
+PERFFLOW_OPTIONS="log-filename-include=instance-path" ../smoketest.py &&
+    sha1=$(echo -n "1" | sha1sum | awk "{print \$1}" | cut -c1-8) &&
+    test -f perfflow.{${sha1}}.pfw &&
+    sanity_check perfflow.{${sha1}}.pfw &&
+    rm perfflow.{${sha1}}.pfw
+'
+
+test_expect_success 'PERFFLOW_OPTIONS: Flux supported' '
+    FLUX_JOB_ID=123456 \
```
Comment (SteVwonder) [inline src/python/test/t0001-pybinding-basic.t:117]: Looks like flux uses the unicode `ƒ` in the env var.  Might be worth testing that that oddity works here:

```
ƒ(s=1,d=0) fluxuser@dc9ae6ac0bde:~$ flux mini run printenv | grep FLUX_JOB_ID
FLUX_JOB_ID=ƒeF9QZG3
```

**Review by SteVwonder — APPROVED** [review]
Thanks @dongahn! Just two small comments

File: src/python/test/t0001-pybinding-basic.t (hunk 2 above)
Comment (dongahn) [reply src/python/test/t0001-pybinding-basic.t:117]: Thanks.

**Pull request merged**